src/Microsoft.Health.Dicom.Core/Features/Query/QueryParser.TagValueParser.cs

```diff
@@ -76,5 +134,25 @@ private static DateTime ParseDate(string date, string tagName)
 
             return parsedDate;
         }
+
+        private static DateTime ParseDateTime(string dateTime, string tagName)
+        {
+            if (!DateTimeOffset.TryParseExact(dateTime, DateTimeTagValueFormats, null, System.Globalization.DateTimeStyles.None, out DateTimeOffset parsedDateTimeOffset))
```

Comment:
Since that are specific formats that we accept (DateTimeTagValueFormats), we don't necessarily need the CultureInfo.InvariantCulture. This is also the case with the already existing ParseDate method.
If you feel like it should be changed to CultureInfo.InvariantCulture, please feel free to respond or reach out offline.